backend/src/routes/index.js

```diff
@@ -23,6 +23,17 @@ async function ensureUser(req, res, next) {
   next()
 }
 
+async function checkSingularity({userId, computerId, sessionId}) {
+  const singularities = await Singularity.find({
```

Comment:
this function should return a promise instead of a resolved value — you should just `return Singularity.find(...` instead of `await`ing and returning the result.